LocalSettings.php

```diff
@@ -1,63 +1,375 @@
+
+/**
+ * @see BAC-1235
+ */
+if ( !empty( $maintClass ) && $maintClass == 'RebuildLocalisationCache' ) {
+    $wgLocalisationCacheConf['manualRecache'] = false;
+}
+
+/**
+ * A collection of general-purpose functions, wf*() family.
+ */
+require_once "$IP/includes/GlobalFunctions.php";
+require_once "$IP/includes/wikia/GlobalFunctions.php";
+
+/**
+ * Launch the profiler.
+ */
+if( !function_exists( 'wfProfileIn' ) ) {
+    require_once "$IP/StartProfiler.php";
+}
+
+/**
+ * Manipulate IEUrlExtension::areServerVarsBad() to work well with our Apache
+ * mod_rewrite rules.
+ * @see IEUrlExtension::areServerVarsBad()
+ * @see includes/WebRequest.php
+ */
+unset( $_SERVER[ 'SERVER_SOFTWARE' ] );
+
+/**
+ * The file is generated by deploy tools upon release. It sets $wgStyleVersion
+ * to proper value.
+ * @see $wgStyleVersion
+ */
+require_once "$IP/wgStyleVersion.php";
+
+/**
+ * Set domains that should be considered local ones.
+ * @see $wgConf
+ * @see includes/HttpFunctions.php
+ */
+$wgConf->localVHosts = array_merge(
+    $wgWikiFactoryDomains,
+    [
+        $wgWikiaBaseDomain, 
+        'uncyclopedia.org',
+        'memory-alpha.org',
+        'wowwiki.com',
+    ]
+);
+
+/**
+ * Load environment-specific settings.
+ * @see https://github.com/Wikia/chef-repo/blob/master/cookbooks/common/attributes/default.rb
+ * @see $wgWikiaEnvironment
+ */
+require_once "$IP/../config/$wgWikiaEnvironment.php";
+
+/**
+ * Example read-only settings.
+ * @see $wgLBFactoryConf
+ * @see includes/db/LBFactory.php
+ * @see includes/api/wikia/ApiFetchBlob.php
+ */
+// $wgLBFactoryConf['readOnlyBySection']['DEFAULT'] = 'We apologise, the site is temporarily in read-only state';
+// $wgLBFactoryConf['readOnlyBySection']['central'] = 'We apologise, this wiki is temporarily in read-only state';
+// $wgLBFactoryConf['readOnlyBySection']['semanticdb'] = 'We apologise, this wiki will be in read-only state till about 15:00 UTC';
+// $wgLBFactoryConf['readOnlyBySection']['c4'] = 'We apologise, this wiki is temporarily in read-only state';
+// $wgReadOnly = 'We apologize, database maintenance in progress. Editing is locked while we update our system.';
+// $wgLBFactoryConf['readOnlyBySection']['mail'] = 'We apologize, database maintenance in progress.';
+// $wgLBFactoryConf['readOnlyBySection']['specials'] = 'We apologize, specials database maintenance in progress.';
 
-/* @var $wgDBcluster string */
-// in some cases $wgMemc is still null at this point, let's initialize it (SUS-2699)
+/**
+ * Calculate configuration for SwiftFileBackend.
+ * @see includes/filerepo/backend/SwiftFileBackend.php
+ * @var Array $wgFSSwiftServer
+ * @var Array $wgFSSwiftConfig
+ */
+$wgFSSwiftServer = $wgFSSwiftDC[$wgWikiaDatacenter]['server'];
+$wgFSSwiftConfig = $wgFSSwiftDC[$wgWikiaDatacenter]['config'];
+
+/**
+ * The URL path of the skins directory. Will default to "{$wgScriptPath}/skins"
+ * in Setup.php.
+ */
+$wgStylePath = "$wgResourceBasePath/skins";
+
+/**
+ * 
```

Comment:
Description is missing here. Proposal: `The URL path of the extensions directory` ;)